### ISSUE TITLE
Galaxy Tours tour_id for repeats and sections

### ISSUE THREAD

**Issue by djcomlab** [galaxyproject/galaxy#6471]
Hello!

I've been trying to create quite a complex Galaxy tour for one of our tools https://github.com/ISA-tools/isatools-galaxy/tree/develop/tools/isacreate where we nest a lot of sections and repeated parameter sections.

While it's quite easy to use the CSS selectors to find the `tour_id` on inputs like input boxes and text boxes etc., unless I am mistaken I have noticed that there are not any references for `tour_id` to allow the tours to expand `<section>` or to get near `<repeat>` section buttons. We originally implemented the tours by referencing the `id` generated on some places (`uid-x` and `field-uid-x`) but these are dynamic so break if the user of the tour adds/removes repeated sections then goes backwards and forwards in the tour (if that makes sense).

Would adding `tour_id` to sections and repeats be something easy enough to implement?

Best/David

**Comment by bgruening** [comment]
I think this is a good idea. Mabye @guerler knows more about this?

**Comment by dannon** [comment]
This is a conversation we've had multiple times already, the plan is to just improve semantic markup of existing classes and deprecate 'tour_id' completely.  It makes things worse to have to maintain multiple layers of markup.

@djcomlab Can you share specific examples of what you're trying to do?  We definitely do want to enhance the existing id/class markup to accommodate things like this  (just not via tour_id).

**Comment by djcomlab** [comment]
OK, fair enough! At the moment I've had to use some very convoluted CSS selectors, but I've got it working somehow anyway.

**Comment by dannon** [comment]
@djcomlab Awesome, glad you got it working!  Anything we can do to clean up those selectors, suggestions you might have, would be very valuable.

What I meant above, more generally, is that anything with a page-unique 'tour_id', should just have that 'tour_id' flattened to the html element id, and we should reconcile any differences there -- picking the 'one true id' to move forward with.  Anything that had a tour_id that wasn't page-unique (beyond probably being currently broken) should have those tour_ids flattened into the current classes of the element.

**Comment by bgruening** [comment]
@djcomlab do you know this one? https://github.com/TailorDev/galaxy-tourbuilder

**Comment by djcomlab** [comment]
@bgruening Yep, this is what I started using! But it only picks out elements where there is a `tour_id`, for example `div[tour_id="study_metadata|title"]` and does not seem to select anything else. It's useful for getting started, but for what we wanted to do it's not quite there yet.

**Comment by bgruening** [comment]
Maybe you can fill an issue at he project. Can this issue be closed? Thanks, looking forward to play with your tour :)

**Comment by dannon** [comment]
@bgruening I'd like to keep this one open.  I've added the id refactoring (as well as creating a tour-loader adapter interface to migrate smoothly) to my TODO list and will close it out when done.  (out of town next week, probably after that)

**Comment by djcomlab** [comment]
@dannon Examples of what I'm trying to do - basically have the tour be able to click and add items when there's a tool `<repeat>` section (requires hitting the button basically), and also the tour to be able to expand an unexpanded tool `<section>`.

The tool I'm working on is https://github.com/ISA-tools/isatools-galaxy/tree/develop/tools/isacreate and the tour https://github.com/ISA-tools/isatools-galaxy/blob/develop/tours/isacreate_metabo_tour.yaml

**Comment by dannon** [comment]
@djcomlab Thanks for that tour example, that's exactly what I was hoping to test against.  I'll take a look!
